frontend/src/components/organisms/NavBar.tsx

```diff
@@ -13,8 +13,7 @@ const NavBar = () => {
   const handleSignOut = async () => {
     try {
       await signOutUser();
-      // clear cache!!
-      queryClient.clear();
+      queryClient.clear(); // Clear cache for react query
```

Comment:
The reason why I did this is relevant. I actually discovered this (bug?) accidentally
- For pagination, each page has its own unique query key using the page number
- I set the staleTime to Infinity so that the data is not updated with background re-fetches (react-query does this). This way, we can check previous and next pages seamlessly
- now when a user signs out and another user signs in they will see the data of the other user that was cached before the actual data is updated so we need to clear the entire cache when the session changes